### ISSUE TITLE
Allow displaying a different field that the linked `valueField` in `Relation` widget

### ISSUE THREAD

**Issue by medfreeman** [netlify/netlify-cms#591]
**- Do you want to request a *feature* or report a *bug*?**

Feature
Needs #597 first.

**- What is the current behavior?**

The [`Relation` widget](https://github.com/netlify/netlify-cms/blob/5b502aa68004118d8036d99f14126e998ddc5533/src/components/Widgets/RelationControl.js) allows linking to another collection item.
The `searchFields` property defines in what item fields netlify will look for the user provided string.
The `valueField` property defines what value will be stored in the relation field, thus creating a link between the current item and searched item.

**- Context**

I created a `menus` collection that can link to `pages` collection items on two levels:

```
  - name: "menus"
    label: "Menu"
    editor:
      preview: false
    folder: "content/menus"
    create: true
    widget: "object"
    fields:
      - { label: "Title", name: "title", widget: "string" }
      - label: "Items"
        name: "items"
        widget: "list"
        fields:
          - { label: "Page", name: "page", widget: "relation", collection: "pages", searchFields: ["title", "body"], valueField: "uuid" }
          - label: "Items"
            name: "items"
            widget: "list"
            fields:
              - { label: "Page", name: "page", widget: "relation", collection: "pages", searchFields: ["title", "body"], valueField: "uuid" }
```

I also created an `uuid` *read-only* widget that allows linking to a page by a *constant* unique per-item field, thus preventing the menu links to break in the case the `valueField` is set to `title` and the user changes the page `title` for example.

**- What is the expected behavior?**

Add a `displayField` property that would allow displaying a different field that the linked `valueField`.

In my above example i would have `uuid` as `valueField` and `title` as `displayField`.
The user would see the page title as menu entry, but the stored value would be the uuid.

**Comment by barthc** [comment]
@tomrutgers please confirm this issue is already fixed in #1303  before closing.

**Comment by tomrutgers** [comment]
Correct. You can now use `displayFields: []`. It defaults to `valueField`

**Comment by papandreou** [comment]
@tomrutgers, `displayFields` only has an effect on the autocompleter, though, not when a related entry has been selected. This means that a relation to a "id" like field will look user unfriendly most of the time:

![screen shot 2019-01-09 at 13 40 22](https://user-images.githubusercontent.com/373545/50900039-2558f880-1414-11e9-8b28-cc26946f7bf8.png)

... I think this is what the OP means.

Also discussed here: https://github.com/netlify/netlify-cms/issues/1982#issuecomment-450687500

**Comment by tomrutgers** [comment]
This issue was opened when there wasn't a `displayFields` property available, so it is a bit different from #1982 I suppose. Since then a `displayFields` has been added but it has it issues for selection.  Would you like to keep both issues open @papandreou?

**Comment by papandreou** [comment]
@tomrutgers, hmm, I think #1982 covers the remaining bits pretty well, so it's fine that this one is closed. Just wanted to clarify :+1:
